CMakeLists.txt

```diff
@@ -16,34 +24,80 @@ if (EXISTS "${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
     conan_basic_setup()
 endif()
 
+#############
+# User config
+#############
+
+include(CMakeDependentOption)
 option(STD_C "Use the standard C library" ON)
 option(STD_CPP "Use the standard C++ library" ON)
-option(CPPUTEST_FLAGS "Use the CFLAGS/CXXFLAGS/LDFLAGS set by CppUTest" ON)
-option(MEMORY_LEAK_DETECTION "Enable memory leak detection" ON)
+option(CPPUTEST_FLAGS "Use the CFLAGS/CXXFLAGS/LDFLAGS set by CppUTest" ${PROJECT_IS_TOP_LEVEL})
+cmake_dependent_option(MEMORY_LEAK_DETECTION "Enable memory leak detection" ${PROJECT_IS_TOP_LEVEL} "STD_C;NOT BORLAND;CPPUTEST_FLAGS" OFF)
 option(EXTENSIONS "Use the CppUTest extension library" ON)
-option(LONGLONG "Support long long" OFF)
-option(MAP_FILE "Enable the creation of a map file" OFF)
-option(COVERAGE "Enable running with coverage" OFF)
-option(C++11 "Compile with C++11 support" OFF)
-option(WERROR "Compile with warnings as errors" OFF)
+cmake_dependent_option(LONGLONG "Support long long" OFF "NOT BORLAND" OFF)
+cmake_dependent_option(MAP_FILE "Enable the creation of a map file" OFF "NOT MSVC;CPPUTEST_FLAGS" OFF)
+cmake_dependent_option(COVERAGE "Enable running with coverage" OFF "NOT MSVC;CPPUTEST_FLAGS" OFF)
+cmake_dependent_option(CPPUTEST_USE_CXX11 "Compile with C++11 support" OFF "CPPUTEST_FLAGS" OFF)
+cmake_dependent_option(WERROR "Compile with warnings as errors" OFF "CPPUTEST_FLAGS" OFF)
 
-option(TESTS "Compile and make tests for the code?" ON)
+option(TESTS "Compile and make tests for the code?" ${PROJECT_IS_TOP_LEVEL})
 option(TESTS_DETAILED "Run each test separately instead of grouped?" OFF)
-option(TESTS_BUILD_DISCOVER "Build time test discover" ON)
-
-option(EXAMPLES "Compile and make examples?" OFF)
-
-option(VERBOSE_CONFIG "Print configuration to stdout during generation" ON)
-
-option(LIBNAME_POSTFIX_BITSIZE "Add architecture bitsize (32/64) to the library name?" OFF)
-option(LIBNAME_POSTFIX_DEBUG "Add indication of debug compilation to the library name?" OFF)
-
-option(HAS_INF "Compiler has Inf value for float" ON)
-option(HAS_NAN "Compiler has NaN value for float" ON)
+cmake_dependent_option(TESTS_BUILD_DISCOVER "Build time test discover" ON "NOT CMAKE_CXX_COMPILER_ID STREQUAL IAR" OFF)
+
+option(EXAMPLES "Compile and make examples?" ${PROJECT_IS_TOP_LEVEL})
+
+option(VERBOSE_CONFIG "Print configuration to stdout during generation" ${PROJECT_IS_TOP_LEVEL})
+
+cmake_dependent_option(LIBNAME_POSTFIX_BITSIZE "Add architecture bitsize (32/64) to the library name?" OFF "PROJECT_IS_TOP_LEVEL" OFF)
+cmake_dependent_option(LIBNAME_POSTFIX_DEBUG "Add indication of debug compilation to the library name?" OFF "PROJECT_IS_TOP_LEVEL" OFF)
+
+cmake_dependent_option(HAS_INF "Compiler has Inf value for float" ON "NOT BORLAND" OFF)
+cmake_dependent_option(HAS_NAN "Compiler has NaN value for float" ON "NOT BORLAND" OFF)
```

Comment:
_(Not directly related to this PR)_ Does it make sense to let CMake detect these?

Reply:
Probably?
- https://en.cppreference.com/w/cpp/numeric/math/NAN
- https://en.cppreference.com/w/cpp/numeric/math/INFINITY

Reply:
That's C++11 (or later) only :-(

Reply:
The reason port to Borland C++ didn't supported NaN and INF properly, so I assume it is still needed.

Reply:
https://github.com/cpputest/cpputest/pull/1584

Reply:
https://github.com/cpputest/cpputest/pull/1585